src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -56,6 +68,82 @@ public synchronized String getCurrentCommandString() {
 			return getCurrentCommand().toString();
 	}
 
+	/**
+	 * Runs current command, if set.
+	 *
+	 */
+
+	public synchronized void runCurrentCommand(){
+		if(currentCommand != null){
+			currentCommand.execute(driverManager.getCurrentDriver());
+		}
+	}
+
+	/**
+	 * Deletes observers and move observers collection to cache
+	 */
+
+	public synchronized void deleteCurrentObservers(){
+
+		if(changePublisher.getSubscribers().size() > 0){
+			cache = new ArrayList<>(changePublisher.getSubscribers());
+		}
+
+		changePublisher.clearObservers();
+	}
+
+	/**
+	 * Restores observers from cache
+	 * NOTE:
+	 * It's probable if addObserver/s feature would be implemented, the cache should be empty
+	 */
+
+	public synchronized void resetObservers(){
+		if(cache != null && cache.size() > 0){
+			cache.forEach(cached -> changePublisher.addSubscriber(cached));
+			clearCache();
+		}
+	}
+
+	/**
+	 * Add new observer for list of subscribers that belongs to <code>changePublisher</code>
+	 * It should clear cache, if exists.
+	 * @param subscriber new observer to add.
+	 */
+	public synchronized void addObserver(Subscriber subscriber){
+		if(subscriber != null){
+			changePublisher.addSubscriber(subscriber);
+		}
+	}
```

Comment:
The idea was to make the publisher available so we do not need to duplicate its API. However, this is all worth considering.

Reply:
Method removed, I came back to previous basic solution in latest commit